facilitator/src/gcp_oauth.rs

```diff
@@ -393,13 +452,14 @@ impl GcpOauthTokenProvider {
     /// The returned value is an owned reference because the token owned by this
     /// struct could change while the caller is still holding the returned token
     fn ensure_default_account_token(&mut self) -> Result<String> {
-        if let Some(token) = &self.default_account_token {
+        if let Some(token) = &*self.default_account_token.read().unwrap() {
```

Comment:
nit: This took me a couple of seconds to understand, it might be nice if this was 

```
&(*self.default_account_token.read().unwrap())
```

or it might be worse, honestly no idea.

Reply:
I tried rewriting it as
```
if let Some(token) = self.default_account_token.read().unwrap().deref().as_ref() { ... }
```
...and I don't think it's anymore clear: if I don't know how `std::sync::RwLock` works, I still have to trawl through `std::sync` docs to understand what `as_ref` does. I also really don't like that I'm forced to insert an `unwrap()` call, which IMO is usually a sign that you're doing something sketchy.

I think Rust programmers just have to get used to using `*` to dereference `RwLock{Read|Write}Guard<T>` and `MutexGuard<T>` into `T`. I do with those types provided an explicit method like `deref_mutex()` that would make this more obvious for those of us who weigh readability over terseness but given the numerous examples in Rustdoc, this form seems to be the least surprising for Rust programmers.